src/library/scanner/scannerglobal.h

```diff
@@ -63,9 +63,9 @@ class ScannerGlobal {
         return m_trackLocations.contains(trackLocation);
     }
 
-    // Returns the directory hash if it exists or -1 if it doesn't.
-    inline mixxx::cache_key_t directoryHashInDatabase(const QString& directoryPath) const {
-        return m_directoryHashes.value(directoryPath, -1);
+    // Returns the directory hash if it exists or mixxx::invalidCacheKey() if it doesn't.
+    mixxx::cache_key_t directoryHashInDatabase(const QString& directoryPath) const {
+        return m_directoryHashes.value(directoryPath, mixxx::invalidCacheKey());
```

Comment:
I think invalidCacheKey() is optimized to a plain 0 here.
So it is the same is the directory is empty, right.
Is this intended?

Did we have an issue with -1? 

We can basically use any magic number and treat it as Invalid.
If we want to add an extra safety net, we may replace -1 by -2 or something wenn a hash is actually of this invalid value, which is unlikely.

What is the hash key of a directory that holds only a  single file "invalidCacheKey.mp3" it such?

Reply:
Magic number -1 **is** the issue here as I explained! It is a valid uin64 hash and all new directories were detected as existing and never inserted into the database.

I will not fix any other issues related to the library scanner.

Reply:
**All** magic "negative" numbers are valid hash keys.

Reply:
It doesn't matter how many files are in a directory. Even empty directories produce a hash <> 0.

Reply:
See: mixxx::cacheKeyFromMessageDigest()

Reply:
If you don't believe me, an empty root directory after the fix:
![Screenshot from 2021-01-25 11-13-36](https://user-images.githubusercontent.com/1474154/105692297-88842d80-5efe-11eb-8667-0ea4467ecacc.png)